Assets/MixedRealityToolkit/Utilities/AssetScriptReferenceRetargeter.cs

```diff
@@ -0,0 +1,593 @@
+                        }
+
+                        if (dllGuids.ContainsKey(file.Name))
+                        {
+                            metaFilePath = $"{file.FullName}.meta";
+                            metaFileContent[1] = $"guid: {dllGuid}";
+                            if (File.Exists(metaFilePath))
+                            {
+                                File.Delete(metaFilePath);
+                            }
+                            File.WriteAllLines(metaFilePath, metaFileContent);
+                        }
+                    }
+                }
+            }
+        }
+
+        private static StringBuilder guidBuilder = new StringBuilder(32);
+        private static string CycleGuidForward(string guid)
+        {
+            guidBuilder.Clear();
+            //Add one to each hexit in the guid to make it unique, but also reproducible
+            foreach (char hexit in guid)
+            {
+                switch (hexit)
+                {
+                    case 'f':
+                        guidBuilder.Append('0');
+                        break;
+                    case '9':
+                        guidBuilder.Append('a');
+                        break;
+                    default:
+                        guidBuilder.Append((char)(hexit + 1));
+                        break;
+                }
+            }
+            return guidBuilder.ToString();
+        }
+
+        private static void RecursiveFolderCleanup(DirectoryInfo folder)
+        {
+            foreach (DirectoryInfo subFolder in folder.GetDirectories())
+            {
+                RecursiveFolderCleanup(subFolder);
+            }
+
+            string fileCheck;
+            FileInfo[] fileList = folder.GetFiles("*");
+            DirectoryInfo[] folderList = folder.GetDirectories();
+            foreach (FileInfo file in fileList)
+            {
+                if (file.Extension.Equals(".meta"))
+                {
+                    fileCheck = file.FullName.Remove(file.FullName.Length - 5);
+                    bool foundMatch = false;
+                    foreach (FileInfo checkFile in fileList)
+                    {
+                        if (checkFile.FullName.Equals(fileCheck))
+                        {
+                            foundMatch = true;
+                            break;
+                        }
+                    }
+                    if (!foundMatch)
+                    {
+                        foreach (DirectoryInfo checkFolder in folderList)
+                        {
+                            if (checkFolder.FullName.Equals(fileCheck))
+                            {
+                                foundMatch = true;
+                                break;
+                            }
+                        }
+                    }
+                    if (!foundMatch)
+                    {
+                        file.Delete();
+                    }
```

Comment:
Nit: I think we can just use a single if (!foundMatch) scope

Reply:
Actually we can't, but I will update it

---
In reply to: [277841425](https://github.com/Microsoft/MixedRealityToolkit-Unity/pull/4051#discussion_r277841425) [](ancestors = 277841425)